tests/testsuite/registry_overlay.rs

```diff
@@ -0,0 +1,161 @@
+//! Tests for local-registry sources.
+
+use cargo_test_support::registry::{Package, RegistryBuilder, TestRegistry};
+use cargo_test_support::{project, Project};
+
+fn setup() -> (TestRegistry, String) {
+    let alt = RegistryBuilder::new().alternative().build();
+    (
+        RegistryBuilder::new().http_index().build(),
+        alt.index_url()
+            .to_file_path()
+            .unwrap()
+            .into_os_string()
+            .into_string()
+            .unwrap(),
+    )
+}
+
+fn simple_project() -> Project {
+    project()
+        .file(
+            "Cargo.toml",
+            r#"
+                [package]
+                name = "foo"
+                version = "0.0.1"
+                edition = "2015"
+                authors = []
+
+                [dependencies]
+                bar = ">= 0.0.0"
+                baz = ">= 0.1.0"
```

Comment:
Why are non-standard version reqs being used?  Chekhov's gun would say that there is something specific about this to the tests but I'm not seeing it.

(ditto for the other package)

Reply:
I think I just copied it from some other test. I'll switch to normal version reqs.